pyccel/ast/cwrapper.py

```diff
@@ -312,6 +289,186 @@
         """
         return self._var
 
+#-------------------------------------------------------------------
+class PyModule(Module):
+    """
+    Class to hold a module which is accessible from Python.
+
+    Parameters
+    ----------
+    *args : tuple
+        See Module.
+
+    external_funcs : iterable of FunctionDef
+        A list of external functions.
+
+    declarations : iterable
+
+    **kwargs : dict
+        See Module.
+
+    See Also
+    --------
+    Module : The super class from which the class inherits.
+    """
+    __slots__ = ('_external_funcs', '_declarations')
+    _attribute_nodes = Module._attribute_nodes + ('_external_funcs', '_declarations')
+    def __init__(self, *args, external_funcs = (), declarations = (), **kwargs):
+        self._external_funcs = external_funcs
+        self._declarations = declarations
+        super().__init__(*args, **kwargs)
+
+    @property
+    def external_funcs(self):
+        """
+        A list of external functions.
+
+        The external functions which should be declared at the start of the module.
+        This is useful for declaring the existence of Fortran functions whose
+        definition and declaration is inaccessible from C.
+        """
+        return self._external_funcs
+
+    @external_funcs.setter
+    def external_funcs(self, funcs):
+        for f in self._external_funcs:
+            f.remove_user_node(self)
```

Comment:
/bot accept
This code is only useful if there are already `external_funcs` set. If this is the case it is very important to remove the user node from the old external_funcs. This is often forgotten so it is better to write it now for future proofing reasons rather than remove it.